src/util/internals/Proxy/Proxy.ts

```diff
@@ -0,0 +1,75 @@
+import type { ProxyTarget } from './types';
+
+export function createProxy<T extends ProxyTarget>(target: T) {
+  return new Proxy(target, {
+    get(target, p) {
+      const value = Reflect.get(target, p);
+      return target.transformValue &&
+        Reflect.getOwnPropertyDescriptor(target, p)?.enumerable
+        ? target.transformValue(
+            {
+              operation: 'get',
+              key: p,
+              value,
+            },
+            target
+          )
+        : value;
+    },
+
+    set(target, p, newValue) {
+      const has = Reflect.has(target, p);
+      const prevValue = Reflect.get(target, p);
+      const descriptor = Reflect.getOwnPropertyDescriptor(target, p);
+      const enumerable = !descriptor || descriptor.enumerable;
```

Comment:
wee can decide to allow even non enumerable props. I allow the deepest overriding there is but that is dangerous